wallet/walletrpc.c

```diff
@@ -146,40 +187,168 @@ static struct command_result *json_withdraw(struct command *cmd,
 	}
 
 	maxheight = minconf_to_maxheight(*minconf, cmd->ld);
-	res = wtx_select_utxos(withdraw->wtx, *feerate_per_kw,
-			       tal_count(withdraw->destination), maxheight);
+	res = wtx_select_utxos((*utx)->wtx, *feerate_per_kw,
+			       tal_count((*utx)->destination), maxheight);
 	if (res)
 		return res;
 
-	if (!bip32_pubkey(cmd->ld->wallet->bip32_base, &changekey,
-			  withdraw->wtx->change_key_index)) {
-		return command_fail(cmd, LIGHTNINGD, "Keys generation failure");
+	if (!amount_sat_eq((*utx)->wtx->change, AMOUNT_SAT(0))) {
+		changekey = tal(tmpctx, struct pubkey);
+		if (!bip32_pubkey(cmd->ld->wallet->bip32_base, changekey,
+				  (*utx)->wtx->change_key_index))
+			return command_fail(cmd, LIGHTNINGD, "Keys generation failure");
 	}
-	txfilter_add_scriptpubkey(cmd->ld->owned_txfilter,
-				  scriptpubkey_p2wpkh(tmpctx, &changekey));
 
-	u8 *msg = towire_hsm_sign_withdrawal(cmd,
-					     withdraw->wtx->amount,
-					     withdraw->wtx->change,
-					     withdraw->wtx->change_key_index,
-					     withdraw->destination,
-					     withdraw->wtx->utxos);
+	(*utx)->tx = withdraw_tx(*utx, (*utx)->wtx->utxos,
+				 (*utx)->destination, (*utx)->wtx->amount,
+				 changekey, (*utx)->wtx->change,
```

Comment:
looks like changekey could be uninitialized here if wtx->change is 0?

```
wallet/walletrpc.c: In function ‘json_prepare_tx’:
wallet/walletrpc.c:203:15: error: ‘changekey’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  (*utx)->tx = withdraw_tx(*utx, (*utx)->wtx->utxos,
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      (*utx)->destination, (*utx)->wtx->amount,
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      changekey, (*utx)->wtx->change,
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      cmd->ld->wallet->bip32_base,
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      &(*utx)->change_outnum);
      ~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
That's quite alright, it's not actually used unless we have a non-zero change amount. But still it's a rather annoying warning that we should address.